csrc/ops/utils.cpp

```diff
@@ -350,11 +350,14 @@ IterDomain* newOutputIterDomain(
             .iter_type(iter_type.value())
             .build();
   } else {
+    if (!iter_type.has_value()){
+      iter_type = IterType::Broadcast;
+    }
```

Comment:
Can you say more about when we might hit a case where iter_type is not null or broadcast?

Reply:
Will add a comment.
Right now, I am checking if this fix works generally, and it looks like we cannot have reduction axis with extent 1. Will look at alternative solutions.